### PR TITLE
NEW mass action line update margin and mark rate propal/commande/facture

### PR DESCRIPTION
**Pull request by atm-jonathan**
# Instructions
On propal, orders, and invoices (client)
On line titles, I have added the ability to bulk edit
The margin rate and the brand rate (taux de marque)

# NEW|New [*update mass margin, brand rate*]
Now, bulk editing the margin and markup rates changes the unit price in the same way as editing a single line
I have added the mass action in the file "htdocs/core/tpl/objectline_title.tpl.php"
And in the card.php files for proposals, orders, and invoices, I have managed the unit price, the calculation of the margin and markup rates, as well as the handling of currency conversions.

### CODE REVIEW

File: htdocs/compta/facture/card.php [hunk 1]
```diff
@@ -2134,7 +2134,7 @@
 			}
 			$result = $object->updateline($line->id, $line->desc, $line->subprice, $line->qty, $remise_percent, $line->date_start, $line->date_end, $tvatx, $line->localtax1_tx, $line->localtax2_tx, 'HT', $line->info_bits, $line->product_type, $line->fk_parent_line, 0, $line->fk_fournprice, $line->pa_ht, $line->label, $line->special_code, $line->array_options, $line->situation_percent, $line->fk_unit, $line->multicurrency_subprice);
 		}
-	} elseif ($action == 'addline' && !GETPOST('submitforalllines', 'alpha') && !GETPOST('submitforallmargins', 'alpha') && $usercancreate) {		// Add a new line
+	} elseif ($action == 'addline' && !GETPOST('submitforalllines', 'alpha') && !GETPOST('submitforallmargins', 'alpha') && !GETPOST('submitforallmark', 'alpha')) {		// Add a new line
```
Comment (eldy) [inline htdocs/compta/facture/card.php:2137]: Why removing test on permission ?

Comment (atm-jonathan) [reply htdocs/compta/facture/card.php]: I have corrected

**Comment by atm-jonathan** [comment]
@eldy  Hello,
Is everything OK?
Thanks

File: htdocs/comm/propal/card.php [hunk 2]
```diff
@@ -978,36 +978,56 @@
 			}
 			$result = $object->updateline($line->id, $line->subprice, $line->qty, $remise_percent, $tvatx, $line->localtax1_tx, $line->localtax2_tx, $line->desc, 'HT', $line->info_bits, $line->special_code, $line->fk_parent_line, 0, $line->fk_fournprice, $line->pa_ht, $line->label, $line->product_type, $line->date_start, $line->date_end, $line->array_options, $line->fk_unit, $line->multicurrency_subprice);
 		}
-	} elseif ($action == 'addline' && GETPOST('submitforallmargins', 'alpha') && GETPOST('marginforalllines') !== '' && $usercancreate) {
+	} elseif ($action == 'addline' && (GETPOST('submitforallmargins', 'alpha')
```
Comment (eldy) [inline htdocs/comm/propal/card.php:981]: The && and the || are on same level
a && b || cc
You must use parenthesis to be clear in which test you expect.
Also i recommand to move the $usercancreate that is twice just after the
`if ($action == 'addline' && $usercancreate && (...  ||  ...))
`

**Pull request closed without merging**